src/NuGet.Core/NuGet.Configuration/Settings/SettingsLoadingContext.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Threading;
+
+namespace NuGet.Configuration
+{
+    public class SettingsLoadingContext : IDisposable
+    {
+        private IList<SettingsFile> _settingsFiles = new List<SettingsFile>();
+        private readonly SemaphoreSlim _semaphore;
+        private bool _isDisposed;
+
+        public SettingsLoadingContext()
+        {
+            _semaphore = new SemaphoreSlim(1, 1);
+        }
+
+        internal SettingsFile GetOrCreateSettingsFile(string filePath)
+        {
+            if (filePath == null)
```

Comment:
Why not?
```C#
if (string.IsNullOrEmpty(filePath))
{
    throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Resources.Argument_Cannot_Be_Null_Or_Empty, nameof(filePath)));
}
```

Note that this exception type is more appropriate for the exception message.